src/document.c

```diff
@@ -462,8 +462,11 @@ FIELD_BULK_INDEXER(numericIndexer) {
 
 FIELD_PREPROCESSOR(geoPreprocessor) {
   // TODO: streamline
-  const char *c = RedisModule_StringPtrLen(field->text, NULL);
-  char *pos = strpbrk(c, " ,");
+  size_t len;
+  char buf[32] = { 0 };
+  const char *str = RedisModule_StringPtrLen(field->text, &len);
+  memcpy(buf, str, len);
```

Comment:
(1) why do we copy?
(2) careful with this 32.

Reply:
We have to copy as not to change the original `argv` strings.
 32 bytes should be plenty. we are looking for `longitude` and `latitude` which together should sum up to 20 chars. Anything beyond this is meaningless.

Reply:
Yes, I missed the `*pos =...`.
I don't see how we can assume something on the user input text - it can be of arbitrary length, can't it?
At least, when we memcpy, we need to limit length to `min(32, len)` and then place a '\0' at the end.

Reply:
Sweet!!